### PR TITLE
LabView like wire types

### PR DESCRIPTION
**Pull request by smahmed776**
Screenshot 2024-03-08 at 7 16 24 PM
Screenshot 2024-03-08 at 7 17 57 PM

### CODE REVIEW

**Review by LatentDream — CHANGES_REQUESTED** [review]
I changed two little thing, just some text box that was not at the right place and two connection that were inverted. 

- **We can't see which node raise an error** when an error occurs, which makes it kinda hard to "debug" and we also can not see the error. Not sure if the regression is from this PR or the clean up one, cc: @39bytes 
- Same for the visual effect of showing which block is currently running. It's not on right now

**Review by LatentDream — DISMISSED** [review]
Previous comment fix here: https://github.com/flojoy-ai/studio/pull/1129

File: src/renderer/stores/project.ts [hunk 1]
```diff
@@ -507,7 +512,27 @@ export const useCreateEdge = () => {
...
+          edge.source === connection.source &&
+          edge.target === connection.target &&
+          edge.sourceHandle === connection.sourceHandle &&
+          edge.targetHandle === connection.targetHandle,
+      );
+      if (edgeExists || !connection.source || !connection.target) {
+        return edges;
+      }
+      const newEdge: Edge<EdgeData> = {
+        id: `${connection.source}->${connection.target}`,
```
Comment (39bytes) [inline src/renderer/stores/project.ts:527]: this id is not unique per edge, if you have two edges from block A to block B then both of them will have the same id, it's probably better to use a uuid (or do something like ``${connection.source}->${connection.target} ${(connection.sourceHandle)->(connection.targetHandle)}``

File: docs/src/utils/tailwind.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import type { TVariant } from "@/types/tailwind";
+
+export const variantClassMap: Record<
+  TVariant,
+  Record<"stroke" | "fill" | "text" | "shadow" | "border", string>
```
Comment (itsjoeoui) [inline docs/src/utils/tailwind.ts:5]: this should be an object instead?

```typescript
type Style = {
    stroke: string
    ...
}
```

**Review by itsjoeoui — CHANGES_REQUESTED** [review]

**Review by itsjoeoui — CHANGES_REQUESTED** [review]
https://github.com/flojoy-ai/studio/assets/25695219/be494bad-5add-4e22-985f-d5ec28efb2a9


Inconsistent resizing behaviour:
- sometimes the frame includes the text label and sometimes it does not (when it is small), this should be consistent

**Review by itsjoeoui — CHANGES_REQUESTED** [review]
https://github.com/flojoy-ai/studio/assets/25695219/29da5b92-743d-4486-a99e-ba8685d19efc

The SVG should also adapt to the scale as you resize it imo, the video is from this reactflow example: https://reactflow.dev/examples/nodes/shapes
I sent the code of this example in Slack a bit ago 👀

File: src/renderer/stores/project.ts (hunk 1 above)
Comment (smahmed776) [reply src/renderer/stores/project.ts]: Nice catch @39bytes . Fixed.

**Review by itsjoeoui — APPROVED** [review]
lgtm

**Pull request merged**